src/AudioIO.cpp

```diff
@@ -142,6 +142,30 @@ AudioIO *AudioIO::Get()
    return static_cast< AudioIO* >( AudioIOBase::Get() );
 }
 
+struct AudioIoCallback::TransportState {
+   TransportState(std::weak_ptr<AudacityProject> wOwningProject,
+      const WaveTrackArray &playbackTracks,
+      unsigned numPlaybackChannels, double rate)
+   {
+      if (auto pOwningProject = wOwningProject.lock();
+          pOwningProject && numPlaybackChannels > 0) {
+         // Setup for realtime playback at the rate of the realtime
+         // stream, not the rate of the track.
+         pRealtimeScope.emplace(move(wOwningProject), rate);
+         // The following adds a new effect processor for each logical track.
+         for (size_t i = 0, cnt = playbackTracks.size(); i < cnt;) {
+            auto vt = playbackTracks[i].get();
+            unsigned chanCnt = TrackList::Channels(vt).size();
+            i += chanCnt; // Visit leaders only
+            pRealtimeScope->AddTrack(
+               vt, std::min(numPlaybackChannels, chanCnt), rate);
+         }
+      }
+   }
+
+   std::optional<RealtimeEffects::InitializationScope> pRealtimeScope;
```

Comment:
could we rename this to pInitializationScope? (mpInitializationScope, actually) just to reduce the confusion in my head :-) 

anyway I see the point of this now, which is to make sure that any SuspensionScope or ProcessingScope will be bracketed by an InitializationScope, and have this guarantee at compile time.

Reply:
ok